### PR TITLE
updates to secret store docs

### PR DESCRIPTION
**Pull request by willtsai**
Thank you for helping make the Radius documentation better!

**Please follow this checklist before submitting:**

- [x] [Read the contribution guide](https://docs.radapp.dev/contributing/docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Some content in the Secret Store documentation was inaccurate, this PR is to correct those.

### Summary
🗑️📝🧑‍🏫

This pull request updates the documentation and examples for the secret store resource schema to match the current support and limitations of Kubernetes Secrets. It removes `valueFrom` objects from the `secretstore.bicep` snippet and adds notes and disclaimers to the `_index.md` file.

> _No more `valueFrom` in the secret store_
> _The schema has changed, the old way is no more_
> _Beware of the limitations of Kubernetes Secrets_
> _The docs will guide you, but they won't protect you from defeat_

### Walkthrough
* Add a note about the current support for Kubernetes Secrets as secret store resources and the future plans for more options (`docs/content/reference/resource-schema/core-schema/secretstore/_index.md`, [link](https://github.com/project-radius/docs/pull/596/files?diff=unified&w=0#diff-44da821a8f488b766ee893029060ad8f2a2931ca358e73ef1decadbd8cc07ce9R8-R9))
* Update the example value for the `resource` property to use the correct format for referencing a Kubernetes Secret (`docs/content/reference/resource-schema/core-schema/secretstore/_index.md`, [link](https://github.com/project-radius/docs/pull/596/files?diff=unified&w=0#diff-44da821a8f488b766ee893029060ad8f2a2931ca358e73ef1decadbd8cc07ce9L31-R33))
* Add a disclaimer and a note about the unsupported `valueFrom` object for Kubernetes Secrets (`docs/content/reference/resource-schema/core-schema/secretstore/_index.md`, [link](https://github.com/project-radius/docs/pull/596/files?diff=unified&w=0#diff-44da821a8f488b766ee893029060ad8f2a2931ca358e73ef1decadbd8cc07ce9L43-R50))
* Remove the `valueFrom` objects from the `data` object in the secret store bicep snippet (`docs/content/reference/resource-schema/core-schema/secretstore/snippets/secretstore.bicep`, [link](https://github.com/project-radius/docs/pull/596/files?diff=unified&w=0#diff-6220c7f8d14f4759d451828dac8dd69e684a46b400fc625af6c7e32c68fcce78L55-R56))

## Issue reference

Fixes: https://github.com/project-radius/radius/issues/2310

### CODE REVIEW

**Review by youngbupark — DISMISSED** [review]
👍

File: docs/content/reference/resource-schema/core-schema/secretstore/snippets/secretstore.bicep [hunk 1]
```diff
@@ -52,16 +52,8 @@ resource existingAppCert 'Applications.Core/secretStores@2022-03-15-privateprevi
...
-        valueFrom: {
-          name: 'tls.crt'
-        }
-      }
-      'tls.key': {
-        valueFrom: {
-          name: 'tls.key'
-        }
-      }
+      'tls.crt': {}
```
Comment (AaronCrawfis) [inline docs/content/reference/resource-schema/core-schema/secretstore/snippets/secretstore.bicep:56]: Can we add comments here on what this reference means?

Comment (willtsai) [reply docs/content/reference/resource-schema/core-schema/secretstore/snippets/secretstore.bicep:56]: How about a comment as such: `// The keys in this object are the names of the secrets in an external secret store`?

File: docs/content/reference/resource-schema/core-schema/secretstore/_index.md [hunk 2]
```diff
@@ -40,10 +42,12 @@ This property is an object to represent key-value type secrets. You define your
 |------|:--------:|-------------|---------|
 | value | y | The value of the secret key. | `'secretString'`
 | encoding | n | The encoding type of the data value (default is `'raw'`). | `'base64'`
-| [valueFrom](#valuefrom) | n | An object to define the secret store `properties.resource` reference | [See below](#valuefrom)
+| [valueFrom](#valuefrom) | n | A reference to an external secret. This field is currently not in use, as it is meant for supporting more types of external secrets in the future. | [See below](#valuefrom)
```
Comment (AaronCrawfis) [inline docs/content/reference/resource-schema/core-schema/secretstore/_index.md:45]: 🔥

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**